cpp/leaky_osh_eval_stubs.h

```diff
@@ -38,12 +40,160 @@ namespace signal_def {
 const int NO_SIGNAL = -1;
 
 inline List<Tuple2<Str*, int>*>* AllNames() {
-  NotImplemented();
+  List<Tuple2<Str*, int>*>* sigs = NewList<Tuple2<Str*, int>*>();
+
+  // These signals are just the standard ones according to signal(7)
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGHUP\0"), SIGHUP));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGINT\0"), SIGINT));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGQUIT\0"), SIGQUIT));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGILL\0"), SIGILL));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGTRAP\0"), SIGTRAP));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGABRT\0"), SIGABRT));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGIOT\0"), SIGIOT));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGBUS\0"), SIGBUS));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGFPE\0"), SIGFPE));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGKILL\0"), SIGKILL));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGUSR1\0"), SIGUSR1));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGSEGV\0"), SIGSEGV));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGUSR2\0"), SIGUSR2));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGPIPE\0"), SIGPIPE));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGALRM\0"), SIGALRM));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGTERM\0"), SIGTERM));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGSTKFLT\0"), SIGSTKFLT));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGCHLD\0"), SIGCHLD));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGCLD\0"), SIGCLD));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGCONT\0"), SIGCONT));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGSTOP\0"), SIGSTOP));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGTSTP\0"), SIGTSTP));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGTTIN\0"), SIGTTIN));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGTTOU\0"), SIGTTOU));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGURG\0"), SIGURG));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGXCPU\0"), SIGXCPU));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGXFSZ\0"), SIGXFSZ));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGVTALRM\0"), SIGVTALRM));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGPROF\0"), SIGPROF));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGWINCH\0"), SIGWINCH));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGIO\0"), SIGIO));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGPOLL\0"), SIGPOLL));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGPWR\0"), SIGPWR));
+  sigs->append(Alloc<Tuple2<Str*, int>>(StrFromC("SIGSYS\0"), SIGSYS));
```

Comment:
glibc >=2.32 deprecated `sys_siglist`, otherwise we could probably just walk that instead of enumerating everything here...